### PR TITLE
feat(bigtable): add table level IAM policy controls

### PR DESCRIPTION
**Pull request by Emar-Kar**
Closes: #9809

### CODE REVIEW

File: bigtable/google/cloud/bigtable/table.py [hunk 1]
```diff
@@ -138,6 +139,73 @@ def name(self):
...
+
+        :rtype: :class:`google.cloud.bigtable.policy.Policy`
+        :returns: The current IAM policy of this table.
+        """
+        table_client = self._instance._client.table_admin_client
+        resp = table_client.set_iam_policy(resource=self.name, policy=policy.to_pb())
+        return Policy.from_pb(resp)
+
+    def test_iam_permissions(self, permissions):
+        """Returns permissions that the caller has on the specified table.
```
Comment (crwilcox) [inline bigtable/google/cloud/bigtable/table.py]: Can we include that this 'tests the permissions'? similar to https://github.com/googleapis/google-cloud-java/pull/6293/files#diff-323cc5589a85618a42238a082db06902R994

File: bigtable/google/cloud/bigtable/table.py [hunk 2]
```diff
@@ -138,6 +139,73 @@ def name(self):
...
+
+        .. literalinclude:: snippets_table.py
+            :start-after: [START bigtable_table_get_iam_policy]
+            :end-before: [END bigtable_table_get_iam_policy]
+
+        :rtype: :class:`google.cloud.bigtable.policy.Policy`
+        :returns: The current IAM policy of this table.
+        """
+        table_client = self._instance._client.table_admin_client
+        resp = table_client.get_iam_policy(resource=self.name)
```
Comment (crwilcox) [inline bigtable/google/cloud/bigtable/table.py]: this sample takes a resource, but this function doesn't.

File: bigtable/google/cloud/bigtable/table.py [hunk 3]
```diff
@@ -138,6 +139,73 @@ def name(self):
             project=project, instance=instance_id, table=self.table_id
         )
 
+    def get_iam_policy(self):
+        """Gets the access control policy for a table.
```
Comment (crwilcox) [inline bigtable/google/cloud/bigtable/table.py]: as authored this `Gets the IAM access control policy for this table`

File: bigtable/google/cloud/bigtable/table.py [hunk 4]
```diff
@@ -138,6 +139,73 @@ def name(self):
...
+
+        :rtype: :class:`google.cloud.bigtable.policy.Policy`
+        :returns: The current IAM policy of this table.
+        """
+        table_client = self._instance._client.table_admin_client
+        resp = table_client.get_iam_policy(resource=self.name)
+        return Policy.from_pb(resp)
+
+    def set_iam_policy(self, policy):
+        """Sets the access control policy on a table. Replaces any
```
Comment (crwilcox) [inline bigtable/google/cloud/bigtable/table.py]: similarly, not a table, this table.

File: bigtable/google/cloud/bigtable/table.py (hunk 2 above)
Comment (Emar-Kar) [reply bigtable/google/cloud/bigtable/table.py]: If I clearly understand the logic, this method creates a call of the `get_iam_policy` method from the `bigtable_table_admin_client`. That is why it is taking the `resource` as an argument. 

image

File: bigtable/google/cloud/bigtable/table.py [hunk 5]
```diff
@@ -138,6 +139,74 @@ def name(self):
...
+
+        :rtype: :class:`google.cloud.bigtable.policy.Policy`
+        :returns: The current IAM policy of this table.
+        """
+        table_client = self._instance._client.table_admin_client
+        resp = table_client.set_iam_policy(resource=self.name, policy=policy.to_pb())
+        return Policy.from_pb(resp)
+
+    def test_iam_permissions(self, permissions):
+        """Tests whether the caller has the given permissions for the specified table.
```
Comment (crwilcox) [inline bigtable/google/cloud/bigtable/table.py]: Also should be change to be 'this table'

**Pull request merged**